gradoop-flink/src/main/java/org/gradoop/flink/io/impl/csv/indexed/IndexedCSVDataSink.java

```diff
@@ -0,0 +1,153 @@
+  }
+
+  @Override
+  public void write(GraphCollection graphCollection) throws IOException {
+    write(graphCollection, false);
+  }
+
+  @Override
+  public void write(LogicalGraph logicalGraph, boolean overwrite) throws IOException {
+    WriteMode writeMode = overwrite ?
+        WriteMode.OVERWRITE : WriteMode.NO_OVERWRITE;
+
+    DataSet<Tuple3<String, String, String>> metaData;
+    if (!reuseMetadata()) {
+      metaData = createMetaData(logicalGraph);
+    } else {
+      metaData = MetaData.fromFile(metaDataPath, getConfig());
+    }
+
+    DataSet<CSVVertex> csvVertices = logicalGraph.getVertices()
+      .map(new VertexToCSVVertex())
+      .withBroadcastSet(metaData, BC_METADATA);
+
+    DataSet<CSVEdge> csvEdges = logicalGraph.getEdges()
+        .map(new EdgeToCSVEdge())
+        .withBroadcastSet(metaData, BC_METADATA);
+
+    if (!getMetaDataPath().equals(metaDataPath) || !reuseMetadata()) {
+      metaData.writeAsCsv(getMetaDataPath(), CSVConstants.ROW_DELIMITER,
+        CSVConstants.TOKEN_DELIMITER, writeMode).setParallelism(1);
+    }
+    csvVertices.output(internalWriteAsIndexedCsv(csvVertices, new Path(getVertexPath()),
+        CSVConstants.ROW_DELIMITER, CSVConstants.TOKEN_DELIMITER, writeMode));
+
+    csvEdges.output(internalWriteAsIndexedCsv(csvEdges, new Path(getEdgePath()),
+        CSVConstants.ROW_DELIMITER, CSVConstants.TOKEN_DELIMITER, writeMode));
+  }
+
+  @Override
+  public void write(GraphCollection graphCollection, boolean overWrite) throws IOException {
+    throw new UnsupportedOperationException(
+      "Writing a graph collection is currently not supported by this data sink");
+  }
+
+  /**
+   * Returns true, if the meta data shall be reused.
+   *
+   * @return true, iff reuse is possible
+   */
+  private boolean reuseMetadata() {
+    return this.metaDataPath != null && !this.metaDataPath.isEmpty();
+  }
+
+  /**
+   * Writes a {@link Tuple} DataSet as CSV file(s) to the specified location with the specified field and line delimiters.<br>
+   * <b>Note: Only a Tuple DataSet can written as a CSV file.</b><br>
+   * For each Tuple field the result of {@link Object#toString()} is written.
+   *
+   * @param dataSet The Tuple that should be writen in the CSV file.
+   * @param filePath The path pointing to the location the CSV file is written to.
+   * @param rowDelimiter The row delimiter to separate Tuples.
+   * @param fieldDelimiter The field delimiter to separate Tuple fields.
+   * @param wm The behavior regarding existing files. Options are NO_OVERWRITE and OVERWRITE.
+   * @param <X> Tuple for indexed CSV file format.
+   * @return An indexed CSV file format for the tuple.
+   *
+   * references to: {@link DataSet#writeAsCsv(String, String, String, WriteMode)}
+   */
+  @SuppressWarnings("unchecked")
+  private <X extends Tuple> IndexedCSVFileFormat<X> internalWriteAsIndexedCsv(
+    DataSet dataSet, Path filePath, String rowDelimiter, String fieldDelimiter,
+    WriteMode wm) {
```

Comment:
I don't like these shortcuts like `wm` ... change it to `writeMode` for better readability